cmd/evm/internal/t8ntool/execution.go

```diff
@@ -185,17 +185,14 @@ func (pre *Prestate) Apply(vmConfig vm.Config, chainConfig *params.ChainConfig,
 		GasLimit:    pre.Env.GasLimit,
 		GetHash:     getHash,
 	}
-	// Save pre verkle tree to build the proof at the end
-	if pre.VKT != nil && len(pre.VKT) > 0 {
-		switch tr := statedb.GetTrie().(type) {
-		case *trie.VerkleTrie:
-			vtrpre = tr.Copy()
-		case *trie.TransitionTrie:
-			vtrpre = tr.Overlay().Copy()
-		default:
-			panic("invalid trie type")
-		}
+
+	switch tr := statedb.GetTrie().(type) {
+	case *trie.VerkleTrie:
+		vtrpre = tr.Copy()
+	case *trie.TransitionTrie:
+		vtrpre = tr.Overlay().Copy()
 	}
+
```

Comment:
I get that this would be ineffective if the tree was not verkle, but we would need at least a comment explaining that, so that whoever reviews this in geth will not wonder what's going on here.